clients/src/main/java/org/apache/kafka/clients/producer/internals/ProducerInterceptors.java

```diff
@@ -107,14 +110,21 @@ public void onAcknowledgement(RecordMetadata metadata, Exception exception) {
     public void onSendError(ProducerRecord<K, V> record, TopicPartition interceptTopicPartition, Exception exception) {
         for (ProducerInterceptor<K, V> interceptor : this.interceptors) {
             try {
+                Headers headers = record != null ? record.headers() : null;
+                if (headers instanceof RecordHeaders && !((RecordHeaders) headers).isReadOnly()) {
+                    // make a copy of the headers to make sure it is read-only
```

Comment:
The copy is required because we don't want to change the state of origin record's headers, right? If so, could you please enhance the comment?

Reply:
yes, that's correct, https://github.com/apache/kafka/pull/17099#discussion_r1749469391

Reply:
how about 339e3bad72?